.github/workflows/rpc-performance-tests.yml

```diff
@@ -91,24 +92,25 @@ jobs:
                                         --json-report ./reports/mainnet/result.json \
                                         --testing-daemon ${servers[i-1]}
     
-
                # Capture test runner script exit status
                perf_exit_status=$?
 
-               # Save test results to a directory with timestamp and commit hash
-               cp -r ${{runner.workspace}}/rpc-tests/perf/reports/mainnet $RPC_PAST_TEST_DIR/mainnet_$(date +%Y%m%d_%H%M%S)_perf_$(git -C ${{runner.workspace}}/silkworm rev-parse --short HEAD)
+               # Preserve test results
+               mv ${{runner.workspace}}/rpc-tests/perf/reports/mainnet/result.json ${{runner.workspace}}/rpc-tests/perf/reports/mainnet/${servers[i-1]}-$method-result.json
                
+               # Prepare historical test results directory
+               mkdir -p $RPC_PAST_TEST_DIR/mainnet_$(date +%Y%m%d_%H%M%S)_perf_$commit
+               mkdir -p $RPC_PAST_TEST_DIR/mainnet_$(date +%Y%m%d_%H%M%S)_perf_$commit_bin
```

Comment:
I suggest using one fixed-name folder for binary reports (i.e. `mainnet_perf_bin`) and keep them just for the _last_ perf test suite execution, in order to avoid need for huge disk on QA runner hosts (typically a complete binary output for one perf workflow execution requires hundreds of gigs).